contracts/interfaces/IStakedTokenDataProvider.sol

```diff
@@ -59,83 +54,67 @@ interface IStakedTokenDataProvider {
-   * @return stkBptData An object with StkBpt data
-   * @return stkBptUserData An object with user data of StkBpt
-   * @return ethPrice The price of ETH, USD denominated (expressed with 8 decimals)
-   */
-  function getAllStakedTokenUserData(address user)
+  function getStakedUserData(
+    address user,
+    address stakedAsset,
+    address oracleAddress
+  )
     external
     view
-    returns (
-      StakedTokenData memory stkAaveData,
-      StakedTokenUserData memory stkAaveUserData,
-      StakedTokenData memory stkBptData,
-      StakedTokenUserData memory stkBptUserData,
-      uint256 ethPrice
-    );
+    returns (StakedTokenData memory stakedTokenData, StakedTokenUserData memory stakedUserData);
 
   /**
-   * @notice Returns user data of Staked Aave
-   * @param user The address of the user
-   * @return stkAaveData An object with StkAave data
-   * @return stkAaveUserData An object with user data of StkAave
+   * @notice Retrieves data for a batch of staked assets given their addresses and the addresses of their associated oracles.
+   * @param stakedTokens An array of addresses representing the staked tokens.
+   * @param oracleAddresses An array of addresses representing the oracles associated with the staked tokens.
+   * @return stakedData An array containing data about the staked assets.
+   * @return prices An array containing the latest price data from the oracles for the staked tokens.
+   * @return ethPrice An array containing the latest ETH price for the staked tokens.
    */
-  function getStkAaveUserData(address user)
+  function getStakedAssetDataBatch(
+    address[] calldata stakedTokens,
+    address[] calldata oracleAddresses
+  )
     external
     view
-    returns (StakedTokenData memory stkAaveData, StakedTokenUserData memory stkAaveUserData);
+    returns (
+      StakedTokenData[] memory stakedData,
+      uint256[] memory prices,
+      uint256[] memory ethPrice
+    );
 
   /**
-   * @notice Returns user data of Staked Bpt Aave
-   * @param user The address of the user
-   * @return stkBptData An object with StkBpt data
-   * @return stkBptUserData An object with user data of StkBpt
+   * @notice Retrieves data for a batch of staked users and their assets given the addresses of staked tokens, associated oracles, and users.
+   * @param stakedTokens An array of addresses representing the staked tokens.
+   * @param oracleAddresses An array of addresses representing the oracles associated with the staked tokens.
+   * @param userAddresses An array of addresses representing the users whose data should be retrieved.
+   * @return stkAaveData An array containing data about the staked assets.
+   * @return stakedUserData An array containing user-specific data about the staked assets.
    */
-  function getStkBptAaveUserData(address user)
+  function getStakedUserDataBatch(
+    address[] calldata stakedTokens,
+    address[] calldata oracleAddresses,
+    address[] calldata userAddresses
```

Comment:
Do you think an array of users is necessary? UI will probably query for 1 single user at a time.